src/fshtypes/Profile.ts

```diff
@@ -1,18 +1,52 @@
 import { Rule } from './rules/Rule';
 import { FshEntity } from './FshEntity';
+import {
+  InsertRule,
+  MappingRule,
+  CaretValueRule,
+  FixedValueRule,
+  FlagRule,
+  CardRule,
+  ContainsRule,
+  ObeysRule,
+  OnlyRule,
+  ValueSetRule,
+  ConceptRule,
+  ValueSetComponentRule
+} from './rules';
 
 export class Profile extends FshEntity {
   id: string;
   parent?: string;
   title?: string;
   description?: string;
   mixins?: string[];
-  rules: Rule[];
+  rules: SdRule[];
 
   constructor(public name: string) {
     super();
     this.id = name; // init same as name
     this.mixins = [];
     this.rules = [];
   }
+
+  ruleIsAllowed(rule: Rule) {
+    return !(
+      rule instanceof InsertRule ||
+      rule instanceof MappingRule ||
+      rule instanceof ValueSetComponentRule ||
+      rule instanceof ConceptRule
+    );
+  }
 }
+
+export type SdRule =
```

Comment:
I'm not sure this is the right place to export `SdRule`, since it is used by both `Profile` and `Extension`.  Perhaps it should be defined and exported somewhere in the `rules` folder?